src/components/application_manager/test/rpc_service_impl_test.cc

```diff
@@ -0,0 +1,832 @@
+}
+
+TEST_F(RPCServiceImplTest, IsAppServiceRPC_PluginWithWrongName_ReturnFalse) {
+  auto source = am::commands::Command::CommandSource::SOURCE_HMI;
+
+  ON_CALL(mock_app_mngr_, GetPluginManager())
+      .WillByDefault(ReturnRef(mock_rpc_plugin_manager_));
+  PluginOpt mock_rpc_plugin_opt(mock_rpc_plugin_);
+  ON_CALL(mock_rpc_plugin_manager_, FindPluginToProcess(kFunctionId, source))
+      .WillByDefault(Return(mock_rpc_plugin_opt));
+  ON_CALL(mock_rpc_plugin_, PluginName()).WillByDefault(Return(""));
+
+  EXPECT_FALSE(rpc_service_->IsAppServiceRPC(kFunctionId, source));
+}
+
+TEST_F(RPCServiceImplTest,
+       IsAppServiceRPC_CommandFactoryIsAbleToProcess_ReturnTrue) {
+  auto source = am::commands::Command::CommandSource::SOURCE_HMI;
+  using namespace application_manager::plugin_manager::plugin_names;
+
+  ON_CALL(mock_app_mngr_, GetPluginManager())
+      .WillByDefault(ReturnRef(mock_rpc_plugin_manager_));
+  PluginOpt mock_rpc_plugin_opt(mock_rpc_plugin_);
+  ON_CALL(mock_rpc_plugin_manager_, FindPluginToProcess(kFunctionId, source))
+      .WillByDefault(Return(mock_rpc_plugin_opt));
+  ON_CALL(mock_rpc_plugin_, PluginName())
+      .WillByDefault(Return(app_service_rpc_plugin));
+  ON_CALL(mock_rpc_plugin_, GetCommandFactory())
+      .WillByDefault(ReturnRef(mock_command_factory_));
+  ON_CALL(mock_command_factory_, IsAbleToProcess(kFunctionId, source))
+      .WillByDefault(Return(true));
+
+  EXPECT_TRUE(rpc_service_->IsAppServiceRPC(kFunctionId, source));
+}
+
+TEST_F(RPCServiceImplTest,
+       UpdateMobileRPCParams_FunctionSchemaIsResettedAndAddedCustomMember) {
+  using namespace ns_smart_device_link::ns_smart_objects;
+  // Get initial state of schema
+  auto function_id = mobile_apis::FunctionID::OnHMIStatusID;
+  auto message_type = mobile_apis::messageType::notification;
+  MobileAPIWithPublicSchemaSetting extended_mobile_api;
+  extended_mobile_api.SetSchema(function_id, message_type);
+
+  auto rpc_service_with_extended_api =
+      std::make_shared<rpc_service::RPCServiceImpl>(
+          mock_app_mngr_,
+          request_controller_,
+          &mock_protocol_handler_,
+          &mock_hmi_handler_,
+          mock_command_holder_,
+          mock_rpc_protection_manager_,
+          hmi_so_factory_,
+          extended_mobile_api);
+
+  CSmartSchema initial_schema;
+  extended_mobile_api.GetSchema(function_id, message_type, initial_schema);
+  auto initial_schema_item = initial_schema.getSchemaItem();
+  // Change state of schema and add custom parameter
+  auto new_function_id = mobile_apis::FunctionID::OnButtonEventID;
+  std::map<std::string, SMember> members;
+  members["custom member"] = SMember();
+  rpc_service_with_extended_api->UpdateMobileRPCParams(
+      new_function_id, message_type, members);
+  CSmartSchema updated_schema;
+  extended_mobile_api.GetSchema(new_function_id, message_type, updated_schema);
+  // Check that schema is changed
```

Comment:
Place this comment closer to line of code it is related to.
Please check all similar places in file

Reply:
Done in commit d5ec8240cfac58e5ea9f7f9ed13f8127a02f94b9